CDP4Reporting.Tests/DataSource/DataSourceTestFixture.cs

```diff
@@ -244,24 +312,96 @@ public void SetUp()
+            this.ed1.ContainedElement.Add(this.eu4);
+
+            this.ed4.ContainedElement.Add(this.eu5);
+
+            this.ed5.ContainedElement.Add(this.eu6);
+
+            this.ed7.Category.Add(this.cat3);
+            this.ed6.ContainedElement.Add(this.eu7);
+
             // Product tree:
-            // ["cat1"] +> "ed1"
-            // ["cat2"] +-+> "ed1.eu12n1"
-            // [      ] | +--> "ed1.eu12n1.eu2n31"
-            // [      ] | +--> "ed1.eu12n1.eu2n32"
-            // [      ] +-+> "ed1.eu12n2"
-            // [      ] | +--> "ed1.eu12n2.eu2n31"
-            // [      ] | +--> "ed1.eu12n2.eu2n32"
-            // ["cat2"] +-+> "ed1.eu12p1"
-            // ["cat3"] | +--> "ed1.eu12p1.eu2p31"
-            // [      ] | +--> "ed1.eu12p1.eu2p32"
-            // [      ] +-+> "ed1.eu12p2"
-            // ["cat3"] | +--> "ed1.eu12p2.eu2p31"
-            // [      ] | +--> "ed1.eu12p2.eu2p32"
+            // ["cat1"] --> "ed1"
+            // ["cat2"] +--> "ed1.eu12n1"
+            // [      ]   +--> "ed1.eu12n1.eu2n31"
+            // [      ]   +--> "ed1.eu12n1.eu2n32"
+            // [      ] + +> "ed1.eu12n2"
+            // [      ]   +--> "ed1.eu12n2.eu2n31"
+            // [      ]   +--> "ed1.eu12n2.eu2n32"
+            // ["cat2"] +--> "ed1.eu12p1"
+            // ["cat3"]   +--> "ed1.eu12p1.eu2p31"
+            // [      ]   +--> "ed1.eu12p1.eu2p32"
+            // [      ] + --> "ed1.eu12p2"
+            // ["cat3"]   +--> "ed1.eu12p2.eu2p31"
+            // [      ]   +--> "ed1.eu12p2.eu2p32"
+            // ["cat2"] +--> "ed1.eu4"
+            // [      ]   +--> "ed1.eu4.eu5"
+            // [      ]     +--> "ed1.eu4.eu5.eu6"
+            // ["cat3"]       +--> "ed1.eu4.eu5.eu6.eu7"
         }
 
         [Test]
-        public void VerifyStructure()
+        public void VerifyStructure1()
+        {
+            // second row structure
+            // ["cat2"] +--> "ed1.eu4"
+            // [      ]   +--> "ed1.eu4.eu5"
+            // [      ]     +--> "ed1.eu4.eu5.eu6"
+            // ["cat3"]       +--> "ed1.eu4.eu5.eu6.eu7"
+            this.VerifyStructure(this.eu4);
+        }
+
+        [Test]
+        public void VerifyStructure2()
+        {
+            // second row structure
+            // ["cat2"] +--> "ed1.eu4"
+            // ["cat2"]   +--> "ed1.eu4.eu5"
+            // [      ]     +--> "ed1.eu4.eu5.eu6"
+            // ["cat3"]       +--> "ed1.eu4.eu5.eu6.eu7"
+
+            this.ed5.Category.Add(this.cat2);
+            this.VerifyStructure(this.eu5);
+        }
+
+        [Test]
+        public void VerifyStructure3()
+        {
+            // second row structure
+            // ["cat2"] +--> "ed1.eu4"
+            // ["cat2"]   +--> "ed1.eu4.eu5"
+            // ["cat2"]     +--> "ed1.eu4.eu5.eu6"
+            // ["cat3"]       +--> "ed1.eu4.eu5.eu6.eu7"
+
+            this.ed5.Category.Add(this.cat2);
+            this.ed6.Category.Add(this.cat2);
```

Comment:
These 4 methods are kind of "ContinueSetupWith" methods. So Setup extensions.

Reply:
What I was trying to get at is that it is generally a bad idea to have test setup be dependent upon the order of execution of tests. The better way of doing this would be to have different branches in the node hierarchy that test different things, rather that incrementally live modifying the test setup from within the tests.

Reply:
I think it is not dependent upon the order of execution of tests now. SetUp runs before every single test.